watertap/costing/watertap_costing_package.py

```diff
@@ -586,6 +629,84 @@ def cost_caoh2_mixer(blk):
             blk.lime_kg_per_day,
         )
 
+    @staticmethod
+    def cost_crystallizer(blk, cost_type=CrystallizerCostType.default):
+        """
+        Function for costing the FC crystallizer by the mass flow of produced crystals.
+
+        Args:
+            cost_type - Option for crystallizer cost function type - volume or mass basis
+        """
+        _make_capital_cost_var(blk)
+
+        if (
+            cost_type == CrystallizerCostType.default
+            or cost_type == CrystallizerCostType.mass_basis
+        ):
+            WaterTAPCostingData.cost_crystallizer_by_crystal_mass(blk)
+        elif cost_type == CrystallizerCostType.volume_basis:
+            WaterTAPCostingData.cost_crystallizer_by_volume(blk)
+        else:
+            raise ConfigurationError(
+                f"{blk.unit_model.name} received invalid argument for cost_type:"
+                f" {cost_type}. Argument must be a member of the CrystallizerCostType Enum."
+            )

```

Comment:
I do test that this error is raised for the other methods here: https://github.com/watertap-org/watertap/blob/main/watertap/costing/tests/test_watertap_costing_package.py

Reply:
Test added.